lua/complementree/sources.lua

```diff
@@ -166,6 +165,109 @@ local function lsp_completedone(completed_item)
   end
 end
 
+--
+
+local os = string.lower(jit.os)
+local is_linux = (os == 'linux' or os == 'osx' or os == 'bsd')
+local os_sep = is_linux and '/' or '\\'
+local os_path = '[' .. os_sep ..'%w+%-%.%_]*$'
+
+M.filepath_matches = function(opts)
+  local relpath = utils.make_relative_path
+  opts = opts or {}
+  local config = {
+    show_hidden = opts.show_hidden or true,
+    ignore_directories = true,
+    max_depth = opts.max_depth or math.huge,
+    relative_paths = opts.relative_paths or false,
+    ignore = '',
+    root_dirs = opts.root_dirs,
+  }
+
+  local function iter_files()
+    local path_stack = vim.fn.reverse(config.root_dirs or { '.' })
+    local iter_stack = {}
+    for _, p in pairs(path_stack) do
+      table.insert(iter_stack, vim.loop.fs_scandir(p))
+    end
+
+    if config.max_depth == 0 then
+      return function()
+        return nil
+      end
+    end
+
+    return function()
+      local iter = iter_stack[#iter_stack]
+      local path = path_stack[#path_stack]
+      while true do
+        local next, type = vim.loop.fs_scandir_next(iter)
+
+        if not next then
+          table.remove(iter_stack)
+          table.remove(path_stack)
+          if #iter_stack == 0 then
+            return nil
+          end
+          iter = iter_stack[#iter_stack]
+          path = path_stack[#path_stack]
+        elseif
+          (vim.startswith(next, '.') and not config.show_hidden) or (#config.ignore > 0 and next:find(config.ignore))
+        then
+          next = nil
+          type = nil
+        else
+          local full_path = path .. os_sep .. next
+          if type == 'directory' then
+            if #iter_stack < config.max_depth then
+              iter = vim.loop.fs_scandir(full_path)
+              path = full_path
+              table.insert(path_stack, full_path)
+              table.insert(iter_stack, iter)
+            end
+            if not config.ignore_directories then
+              return full_path
+            end
+          else
+            return full_path
+          end
+        end
+      end
+    end
+  end
+
+  return cached('filepath', function(line_to_cursor, _)
+    local pref_start = line_to_cursor:find(os_path)
+    local prefix = line_to_cursor:sub(pref_start)
+
+    -- local items = {}
+    local items = {}
+    for path in iter_files() do
+      items[#items + 1] = path
+    end
+
+    local cwd = vim.fn.getcwd()
+    local display_path
+    local matches = {}
+    for _, path in ipairs(items) do
+      display_path = config.relative_paths and relpath(path, cwd) or path
+
+      matches[#matches + 1] = {
+        word = path,
+        abbr = display_path,
```

Comment:
So you will display the relative path but complete using the full path ? isn't that a bit nasty :wink: